DBM-ChamberOfAspects/Ruby/Zarithrian.lua

```diff
@@ -10,44 +10,65 @@ mod:RegisterEvents(
 	"SPELL_CAST_START",
 	"SPELL_AURA_APPLIED",
 	"SPELL_AURA_APPLIED_DOSE",
+	"SPELL_AURA_REMOVED",
 	"CHAT_MSG_MONSTER_YELL"
 )
 
 local warningAdds				= mod:NewAnnounce("WarnAdds", 3, 74398)
 local warnCleaveArmor			= mod:NewStackAnnounce(74367, 2, nil, "Tank|Healer")
+local warnFearSoon              = mod:NewSoonAnnounce(74384, nil, nil, nil, 2, 2)
 
 local specWarnFear				= mod:NewSpecialWarningSpell(74384, nil, nil, nil, 2, 2)
 local specWarnCleaveArmor		= mod:NewSpecialWarningStack(74367, nil, 2, nil, nil, 1, 6)--ability lasts 30 seconds, has a 15 second cd, so tanks should trade at 2 stacks.
 
-local timerAddsCD				= mod:NewTimer(45.5, "TimerAdds", 74398, nil, nil, 1)
+local timerAddsCD				= mod:NewTimer(40, "TimerAdds", 74398, nil, nil, 1)
 local timerAddsTravel			= mod:NewTimer(10, "AddsArrive") -- Timer to indicate when the summoned adds arive
 local timerCleaveArmor			= mod:NewTargetTimer(30, 74367, nil, "Tank|Healer", nil, 5)
-local timerFearCD				= mod:NewCDTimer(37, 74384, nil, nil, nil, 2)--anywhere from 35-40 seconds in between fears.
+local timerFearCD				= mod:NewCDTimer(30, 74384, nil, nil, nil, 2)
+
+mod:AddBoolOption("CancelBuff")
+local CleaveArmorTargets = {}
 
 function mod:OnCombatStart(delay)
-	timerFearCD:Start(14-delay)--need more pulls to verify consistency
-	timerAddsCD:Start(15.5-delay)--need more pulls to verify consistency
+    timerFearCD:Start(14-delay)
+    warnFearSoon:ScheduleVoice(11, "fearsoon") -- 3 secs prewarning
+    timerAddsCD:Start(18-delay)
+end
+
+function mod:OnCombatEnd()
+    table.wipe(CleaveArmorTargets)
 end
 
 function mod:SPELL_CAST_START(args)
-	if args.spellId == 74384 then
-		specWarnFear:Show()
-		specWarnFear:Play("fearsoon")
-		timerFearCD:Start()
-	end
+    if args.spellId == 74384 then
+        specWarnFear:Show()
+        warnFearSoon:ScheduleVoice(27, "fearsoon") -- 3 secs prewarning
+        timerFearCD:Start()
+    end
 end
 
 function mod:SPELL_AURA_APPLIED(args)
-	if args.spellId == 74367 then
+    if args.spellId == 74367 then
+		if not tContains(CleaveArmorTargets, args.destName) then
```

Comment:
```suggestion
		if self.Options.CancelBuff and not tContains(CleaveArmorTargets, args.destName) then
```
Let's add this check to avoid building the table if we don't have to